### DISCUSSION TITLE
Yarn upgrade equivalent?

### DISCUSSION THREAD

**Discussion by Nantris** [yarnpkg/berry#6582] [Q&A]
`yarn up` and `yarn upgrade-interactive` are both not replacements for the `yarn upgrade` command from Yarn 1.x. Is there really no equivalent? It seems crazy I've spent half and hour trying to find an equivalent command and determined there isn't one?

**Comment by Nantris** [comment]
Okay I see I needed to use `yarn upgrade "**"` (wrapping the asterisks in quotes) but then it still fails if you have a github-based dependency...

> ➤ YN0027: my-package@https://github.com/user/my-package.git can't be resolved to a satisfying range
>
> Internal Error: Package not found

**Comment by Nantris** [comment]
After two hours I'm forced to conclude there's no approach to this except to use `yarn upgrade-interactive` which is beyond tedious when you have dozens of packages and want to keep them up to date on a week-by-week basis.... God forbid someone use a `github:` dependency, then their time is apparently worth nothing to the Yarn dev team.

Sorry for my bitterness, but I spent a lot of time reading through a lot of issues trying to resolve this after spending a lot of time "upgrading" to Yarn Berry to begin with, and the attitude of the dev team to those who shared similar concerns is, frankly, insulting. I could really not be more disappointed. We waited half a decade to move to Yarn Berry to avoid early adopter problems just to find that there's no consideration given to half a dozen use cases that were supported in Yarn 1.x.

I guess I finally have a good reason to check out pnpm.

**Reply by arcanis** [reply]
Yarn is an open-source project; as such, we have limited time to work on it, and only do so because we gain something from it - be it because we're using it in our professional life, because some tasks appeal our personal interest, or simply because we want to help someone nice and have some free time on our hand.

> the attitude of the dev team to those who shared similar concerns is, frankly, insulting

How do you qualify your behaviour? Let me be clear: if you strongly care about a feature, you can work on it, it's the miracle of open-source. We have a [contributing guide](https://yarnpkg.com/advanced/contributing), and frankly the Yarn codebase isn't particularly difficult to grasp. If you prefer to "wait half a decade" in the hope that someone else will do just what you want, that's really on you. Not on a team of volunteers.
